core/pom.xml

```diff
@@ -89,7 +84,6 @@
         <dependency>
             <groupId>com.google.guava</groupId>
             <artifactId>guava</artifactId>
-            <scope>test</scope>
```

Comment:
I added some additional tests and found that my by-hand implementation was naive--I switched back to TypeToken for correctness until I can come up with a correct hand-written implementation.